### PR TITLE
Added essence level edges and snags

### PR DESCRIPTION
**Pull request by JFranklin3000**
Closes [issue URL]

##### In this PR
- Added an option to add edges and snags to all skills in an Essence

##### Testing
- create effect with system.essenceShifts.<essence>.edge or snag validate

### CODE REVIEW

File: module/dice.mjs [hunk 1]
```diff
@@ -91,9 +91,20 @@ export class Dice {
...
+
+    if (actorSkillData.snag || essenceShifts[rolledEssence].snag) {
+      calculatedSnag = true;
+    }
+
     const skillDataset = {
-      edge: actorSkillData.edge,
-      snag: actorSkillData.snag,
+      edge: calculatedEdge,
+      snag: calculatedSnag,
```
Comment (phildominguez) [inline module/dice.mjs:107]:
```suggestion
      edge: actorSkillData.edge || essenceShifts[rolledEssence].edge,
      snag: actorSkillData.snag || essenceShifts[rolledEssence].snag,
```

File: module/dice.mjs [hunk 2]
```diff
@@ -91,9 +91,20 @@ export class Dice {
...
+    let calculatedSnag = false;
+
+    if (actorSkillData.edge || essenceShifts[rolledEssence].edge) {
+      calculatedEdge = true;
+    }
+
+    if (actorSkillData.snag || essenceShifts[rolledEssence].snag) {
+      calculatedSnag = true;
+    }
+
```
Comment (phildominguez) [inline module/dice.mjs:104]:
```suggestion
```

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/dice.mjs (hunk 1 above)
Comment (JFranklin3000) [reply module/dice.mjs]: changed

File: module/dice.mjs (hunk 2 above)
Comment (JFranklin3000) [reply module/dice.mjs]: changed

**Review by phildominguez — APPROVED** [review]

**Pull request merged**